src/incremental/compareAST.ml

```diff
@@ -110,9 +134,10 @@ and eq_enuminfo (a: enuminfo) (b: enuminfo) =
   eq_list eq_eitems a.eitems b.eitems
 (* Ignore ereferenced *)
 
-and eq_args (a: string * typ * attributes) (b: string * typ * attributes) = match a, b with
-    (name1, typ1, attr1), (name2, typ2, attr2) -> name1 = name2 && eq_typ typ1 typ2 && eq_list eq_attribute attr1 attr2
+and eq_args (acc: (typ * typ) list) (a: string * typ * attributes) (b: string * typ * attributes) = match a, b with
+    (name1, typ1, attr1), (name2, typ2, attr2) -> name1 = name2 && eq_typ_acc typ1 typ2 acc && eq_list eq_attribute attr1 attr2
 
+(* TODO: why unused? *)
 and eq_typsig (a: typsig) (b: typsig) =
```

Comment:
`eq_typsig` can be removed. It does not suffice to compare `typsig`s (instead of types) in incremental runs, as the `typsig`s e.g. do not contain the fields of unions and structs.